src/store/services/product.ts

```diff
@@ -0,0 +1,40 @@
+import { IPageData } from 'interface/pageDataInterface'
+import { ICardProduct, IProduct } from 'interface/productInterface'
+
+import { api } from './api'
+
+export const productApi = api.injectEndpoints({
+  endpoints: (builder) => ({
+    getSomeNewProducts: builder.query<ICardProduct[], void>({
+      query: () => ({
+        url: 'product/new'
+      }),
+      providesTags: () => [{ type: 'Product' }]
+    }),
+    getAllProductsByCategory: builder.query<IPageData<ICardProduct[]>, string>({
+      query: category => `product/${category}`,
+      providesTags: (arg) => [{ type: 'Product', category: arg }]
+    }),
+    getSomeProductsByCategory: builder.query<ICardProduct[], string>({
+      query: category => `product/some-category/${category}`,
```

Comment:
`some-category`? it's sound very strange

Reply:
fixed